src/setup-beam.js

```diff
@@ -15,19 +15,24 @@ async function main() {
 
   const osVersion = getRunnerOSVersion()
   const otpSpec = core.getInput('otp-version', { required: true })
-  const otpVersion = await installOTP(otpSpec, osVersion)
-
   const elixirSpec = core.getInput('elixir-version', { required: false })
-  const elixirInstalled = await maybeInstallElixir(elixirSpec, otpVersion)
-  if (elixirInstalled === true) {
-    const shouldMixRebar = core.getInput('install-rebar', {
-      required: false,
-    })
-    await mix(shouldMixRebar, 'rebar')
-    const shouldMixHex = core.getInput('install-hex', {
-      required: false,
-    })
-    await mix(shouldMixHex, 'hex')
+
+  if (otpSpec !== 'false') {
+    const otpVersion = await maybeInstallOTP(otpSpec, osVersion)
+    const elixirInstalled = await maybeInstallElixir(elixirSpec, otpVersion)
+
+    if (elixirInstalled === true) {
+      const shouldMixRebar = core.getInput('install-rebar', {
+        required: false,
+      })
+      await mix(shouldMixRebar, 'rebar')
+      const shouldMixHex = core.getInput('install-hex', {
+        required: false,
+      })
+      await mix(shouldMixHex, 'hex')
+    }
+  } else if (elixirSpec) {
```

Comment:
This should probably be related to Gleam, not Elixir (eg. `!gleamSpec`, or something), because the condition is "fail when no OTP if not Gleam" (otherwise every time we had a new language (eg. LFE) we'd have to come back to this.

Error can be something like "otp-version=false is only available when targeting Gleam".

Reply:
I added the error here to point out to the user that if they had put in both `otp-version` as `false` and `elixir`, `elixir` was not installed (because determining the version currently depends on `otp-version` being provided). So this is only specific to `elixir` and not gleam/lfe/etc.

Reply:
Sure, but we want to warn the other way around (if you chose `otp-version: false` you can only target Gleam). Otherwise, tomorrow we include LFE in the action (and then Caramel, and then...) and have to touch that `if` again. I prefer the error to change to something like I stated "otp-version=false is only available for Gleam".

Reply:
So you're saying there's a higher chance that other languages will have a "sub-version" dependency on OTP similar to Elixir than having the compiler written e.g. in Rust or OCaml.

I am not sure that will be the case. Maybe I'll just remove that `else` and error completely?

(Currently Caramel and Sesterl both are written in OCaml so if I make the change, and someone adds them, and wants to be able to type check without installing OTP, they will need to be added to the new `if` you're proposing)

Reply:
I made the change you're proposing anyway :)

Reply:
No worries, I personally am not even sure this is worth merging if it's a special feature for Gleam only. But maybe Elixir could have a default fallback to oldest "compiled with" `otp-version` if not provided, then it could be useful as well for people using custom BEAM builds as well.